### DISCUSSION TITLE
To PID or not to PID

### DISCUSSION THREAD

**Discussion by garywhiteford** [apache/airflow#21502] [Q&A] [answered]
Why does the Web server example systemd service file include the PID ...

https://github.com/apache/airflow/blob/a2abf663157aea14525e1a55eb9735ba659ae8d6/scripts/systemd/airflow-webserver.service#L29

... but the Scheduler does not?

https://github.com/apache/airflow/blob/a2abf663157aea14525e1a55eb9735ba659ae8d6/scripts/systemd/airflow-scheduler.service#L29

Follow-up question: What happens if I do include a PID file in my systemd file for the Scheduler? Nothing? Or something bad? 

Reference: https://github.com/apache/airflow/pull/855#issuecomment-176603323

**Comment by potiuk** [comment, marked as answer]
I think mostly because it has not been updated after we added the option to run multiple schedulers :).

Before that "There could be only one" anyway.  And running several webserver processes is not unheard of (though it really only makes sense when they run on different machines, because gunicorn can spawn multiple processes anyway). 

But (especially if you have systemd installation) - feel free to update those examples as you see fit. The systemd example is just that - example. 

I think when you add the option, it will burn the memory of your machine and it will heat to the level that it will burn the basement it stands on and open the hole through the Earth crust to the core ;) . 

But Srsly, I think the important is that you use different pid files for every instance. That should be about all that you should worry about .
